saylua/modules/pets/models/db.py

```diff
@@ -1,38 +1,57 @@
-from google.appengine.ext import ndb
+from saylua import db
 
 
 # This is to store alternate linart versions of the same pets
-class SpeciesVersion(ndb.Model):
-    name = ndb.StringProperty()
-    base_image = ndb.StringProperty()
-    base_psd = ndb.StringProperty()
-    default_image = ndb.StringProperty()
+class SpeciesVersion(db.Model):
+
+    __tablename__ = "species_versions"
+
+    id = db.Column(db.Integer, primary_key=True)
+    species_name = db.Column(db.String, db.ForeignKey("species.name"))
+    name = db.Column(db.String(80), unique=True)
+    base_image = db.Column(db.String(200))
+    base_psd = db.Column(db.String(200))
+    default_image = db.Column(db.String(200))
+    species = db.relationship("Species", back_populates="species_versions")
 
 
 # Pets are divided into species and species are divided into variations
-class Species(ndb.Model):
-    name = ndb.StringProperty()
-    versions = ndb.StructuredProperty(SpeciesVersion)
-    description = ndb.TextProperty()
+class Species(db.Model):
+
+    __tablename__ = "species"
+
+    name = db.Column(db.String, primary_key=True)
+    description = db.Column(db.Text)
+    versions = db.relationship("SpeciesVersion", back_populates="species")
+
+
+class SpeciesVariation(db.Model):
+
+    __tablename__ = "species_variations"
+
+    id = db.Column(db.Integer, primary_key=True)
+    species_name = db.Column(db.String, db.ForeignKey("species.name"))
+    name = db.Column(db.Text)
+    description = db.Column(db.Text)
+
 
+class Pet(db.Model):
 
-class SpeciesVariation(ndb.Model):
-    species_id = ndb.IntegerProperty()
-    name = ndb.StringProperty()
-    description = ndb.TextProperty()
+    __tablename__ = "pets"
 
+    id = db.Column(db.Integer, primary_key=True)
+    soulname = db.Column(db.String(80), unique=True)
 
-class Pet(ndb.Model):
-    pet_id = ndb.StringProperty()
-    owner_id = ndb.IntegerProperty()
-    variation_key = ndb.KeyProperty() # Only set if the pet is a variation
-    species_name = ndb.StringProperty() # Note the denormalization
+    owner_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+    # Only set if the pet is a variation
+    variation_key = db.Column(db.Integer, db.ForeignKey("users.id"))
```

Comment:
coat. Also remember to use _id instead of _key now. Also, make this not a foreign key of user.